beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -41,6 +41,16 @@ func (s *Service) validateAggregateAndProof(ctx context.Context, pid peer.ID, ms
 		return pubsub.ValidationIgnore, nil
 	}
 
+	// We should not attempt to process this message if the node is running in optimistic mode.
+	// We just ignore in p2p so that the peer is not penalized.
+	status, err := s.cfg.chain.IsOptimistic(ctx)
+	if err != nil {
+		return pubsub.ValidationReject, nil
```

Comment:
you should return the error here, we have gossip message handler which will log errors for rejected messages. So we make sure that any errors are correctly bubbled up.